### ISSUE TITLE
Very high memory usage

### ISSUE THREAD

**Issue by ghost** [DevBlocky/forceps#1]
```
pub async fn main() {
    let cache = Cache::new("./.cache")
        .build()
        .await
        .unwrap();
    std::thread::sleep(Duration::from_secs(100));
}
```

This program will consume over 21mb of memory.

**Comment by DevBlocky** [comment]
Very odd. Probably related to the `sled` metadata db? I'll test it out tomorrow.

**Comment by DevBlocky** [comment]
Yup, 100% related to `sled`, code causes the same problem:

```rust
fn main() {
    let db = sled::Db::open("./test");
    std::thread::sleep(Duration::from_secs(5));
}
```

It would probably be good to eventually move away from sled (mainly because they don't promise stability yet), however that is more of a long-term goal. Also, I don't really see 21MB of extra RAM usage being that much of a concern.

Out of curiosity, what workload would you plan to use this in where 21MB of RAM usage would be considered extreme and restrict you from using this?

**Comment by ghost** [comment]
21mb is not trivial in rust land 🙂. [This](https://github.com/TatriX/realworld-rust-rocket) service uses less than 4 mb in debug build. forceps alone tripled the memory usage of my application, which loads a bunch of files in memory at boot so most of that is just raw data.

I just looked at the implementation and sled is used only for key value lookup. I'll probably be making a pull request removing sled in a week or so.

**Comment by ghost** [comment]
You might want to notify the upstream maintainers about it. Maybe we won't have to do anything if they can fix it on their end.

**Comment by DevBlocky** [comment]
Makes sense in that context. PRs are completely welcome, and it would be highly appreciated if you could make `read_metadata` and `metadata_iter` async with Future/Streams.

I'll see about searching/making an issue on the sled repository to see what they have to say, and if there is going to be any changes in their next releases.

**Comment by DevBlocky** [comment]
I'm gonna close this because it seems `sled` is no longer being actively developed, and a few MBs of baseline memory usage is not a big deal for most use cases
